firedrake/nullspace.py

```diff
@@ -257,7 +273,7 @@ def _apply(self, matrix_or_ises, transpose=False):
                 raise RuntimeError("Shape of matrix (%d, %d) does not match size of nullspace %d" %
                                    (rows, cols, len(self)))
             # Hang the expanded nullspace on the big matrix
-            self._apply_monolithic(matrix, transpose=transpose)
+            self._apply_monolithic(matrix, transpose=transpose, near=near)
             return
         ises = matrix_or_ises
```

Comment:
In the `near=True` case, you near to do the same thing as currently, composing the split space with the IS.  But you should use the key `"nearnullspace"`.  So do something like:

```python
key = "nearnullspace" if near else "nullspace"
...
   is_.compose(key, ...)
```